### DISCUSSION TITLE
Dockerize demo or other instead of symfony cli

### DISCUSSION THREAD

**Discussion by cavasinf** [kevinpapst/TablerBundle-Demo#7] [General]
As I have no privileges on my pc, I cannot install and use symfony cli to launch/test the demo app.

I've already tried to implement my docker file to launch the project but I've an error when using it. (404 not found I believe)

How should the project works out side of the symfony server ? For example, I don't see any yarn.lock.
How to build the css or js files from webpack config l if there's none.

I'm short in time during this month but I can try to make a docker compose file that allows docker users to test this demo.

**Comment by kevinpapst** [comment]
There is no need to compile anything. 
The demo is meant to be a showcase of the plain bundle without any customization.
It should only use existing features of the bundle, showcase layout and macro options.

Aside from that, it is a plain symfony app that serves from `public/index.php`. 
Run `composer install` and everything should be ready to use.
You can share your docker file and I will have a look (but I am really not good with it).

**Reply by cavasinf** [reply]
I'll try again tomorrow, I've already done a `composer install`..

Maybe a `php bin/console router:debug` will tell me what's wrong.

**Reply by kevinpapst** [reply]
There should be a redirect to `/en/` or `/de/`. Maybe that fails?
It is not something simple as a missing mod_rewrite?

**Reply by cavasinf** [reply]
You are right.
Missing `.htaccess` file in `/public`.

Create new PR `docker-compose`

**Comment by cavasinf** [comment]
So I still have a `404` when `_locale` is auto at `fr` for me. 
Switching manually to `en` doesn't change anything.

![image](https://user-images.githubusercontent.com/25293190/157417885-902b44ee-ad2d-4de7-b3c7-4578974728b0.png)

Here's `php bin/console debug:router`.
```
 -------------------------- -------- -------- ------ ------------------------------------ 
  Name                       Method   Scheme   Host   Path                                
 -------------------------- -------- -------- ------ ------------------------------------ 
  _wdt                       ANY      ANY      ANY    /_wdt/{token}                       
  _profiler_home             ANY      ANY      ANY    /_profiler/                         
  _profiler_search           ANY      ANY      ANY    /_profiler/search                   
  _profiler_search_bar       ANY      ANY      ANY    /_profiler/search_bar               
  _profiler_phpinfo          ANY      ANY      ANY    /_profiler/phpinfo                  
  _profiler_search_results   ANY      ANY      ANY    /_profiler/{token}/search/results   
  _profiler_open_file        ANY      ANY      ANY    /_profiler/open                     
  _profiler                  ANY      ANY      ANY    /_profiler/{token}                  
  _profiler_router           ANY      ANY      ANY    /_profiler/{token}/router           
  _profiler_exception        ANY      ANY      ANY    /_profiler/{token}/exception        
  _profiler_exception_css    ANY      ANY      ANY    /_profiler/{token}/exception.css    
  homepage                   ANY      ANY      ANY    /{_locale}/                         
  forms                      ANY      ANY      ANY    /{_locale}/forms                    
  buttons                    ANY      ANY      ANY    /{_locale}/buttons                  
  dropdown                   ANY      ANY      ANY    /{_locale}/dropdown                 
  alert                      ANY      ANY      ANY    /{_locale}/alert                    
  timeline                   ANY      ANY      ANY    /{_locale}/timeline                 
  full-page                  ANY      ANY      ANY    /{_locale}/full-page                
  documentation              ANY      ANY      ANY    /{_locale}/documentation/{chapter}  
  error403                   ANY      ANY      ANY    /{_locale}/error-403                
  error404                   ANY      ANY      ANY    /{_locale}/error-404                
  error500                   ANY      ANY      ANY    /{_locale}/error-500                
  forms-horizontal           ANY      ANY      ANY    /{_locale}/forms/horizontal         
  profile                    ANY      ANY      ANY    /{_locale}/profile                  
  dark-mode                  ANY      ANY      ANY    /{_locale}/dark-mode                
  light-mode                 ANY      ANY      ANY    /{_locale}/light-mode               
  navbar-overlapping         ANY      ANY      ANY    /{_locale}/navbar-overlapping       
  security_login             ANY      ANY      ANY    /{_locale}/login                    
  security_logout            ANY      ANY      ANY    /{_locale}/logout                   
 -------------------------- -------- -------- ------ ------------------------------------ 
```


 ----------------------

I was surprised by the `.env.dist` file.
`public/index.php` looks old (same for `bin/console`).
No `symfony/runtime` dependency ?

Is this an OLD symfony 4 project updated in symfony 5 ?

**Reply by cavasinf** [reply]
I've updated the file to symfony 5.x in a new PR `symfony-5`
